fix(c/sql): stored procedure/function metadata

zregvart opened the pull request:
When fetching metadata for functions in PostgreSQL we need to use stored
procedure metadata, not function metadata as it returns nothing.

Ref. https://issues.redhat.com/browse/ENTESB-16358

fix(c/sql): make sure test tear down is done (766e285)

We need to make sure that any object we create is dropped in the test.
Otherwise we might get flaky tests that depend on the order of
execution.

(cherry picked from commit fbcbfeac2a2111e7f43f60458b88a7472bff5778)

zregvart committed:
fix(c/sql): make sure test tear down is done

We need to make sure that any object we create is dropped in the test.
Otherwise we might get flaky tests that depend on the order of
execution.

(cherry picked from commit fbcbfeac2a2111e7f43f60458b88a7472bff5778)

zregvart committed:
fix(c/sql): stored procedure/function metadata

When fetching metadata for functions in PostgreSQL we need to use stored
procedure metadata, not function metadata as it returns nothing.

Ref. https://issues.redhat.com/browse/ENTESB-16358

zregvart commented:
retest this please

avano commented:
retest this please

claudio4j commented on app/connector/sql/src/main/java/io/syndesis/connector/sql/SqlSupport.java:
```diff
@@ -49,15 +47,13 @@ public static Connection createConnection(final Map<String, Object> properties)
     }
 
     public static StoredProcedureMetadata getFunctionOrProcedureMetadata(final Connection connection, final String catalog,
```
As there is no fetchFunctionColumns call, renaming the method back to getProcedureMetadata and adding a comment about why not using fetchFunctionColumns.

zregvart committed:
refactor(c/sql): stored procedure metadata cleanup

Renamed `SqlSupport::getFunctionOrProcedureMetadata` to
`SqlSupport::getProcedureMetadata` to reflect what we support. Added
tests to assert supported scenarios.

zregvart committed:
Delete dependabot.yml

zregvart commented:
Thanks for the review, @claudio4j can you have another look?
